### DISCUSSION TITLE
可以请教一下怎么找的基地址吗

### DISCUSSION THREAD

**Discussion by 204065248** [FrozenFish259/Goose_Goose_Duck_Hack#115] [Q&A 疑难解答区] [answered]
如题

**Comment by FrozenFish259** [comment, marked as answer]
用CE开启Mono功能，搜类名找到类的Class地址，然后搜Class地址就能找到唯一的静态变量基址

**Comment by 204065248** [comment]
为啥我的Ce开启mono后，里面没内容呢

**Comment by FrozenFish259** [comment]
![image](https://user-images.githubusercontent.com/49237108/212536708-96b6f09d-4b56-4711-b876-cb8ebc61e105.png)

**Comment by 204065248** [comment]
哦哦，我看的是另一个，吃完饭看看这个有没有

**Comment by 204065248** [comment]
![image](https://user-images.githubusercontent.com/17307843/212541950-9cad345d-5c22-4a74-9c4b-980a3228887f.png)
w
我这里是空的。。，请问一下你是怎么搞的

**Comment by FrozenFish259** [comment]
Search

**Comment by 204065248** [comment]
这个我知道，主要是我里面没有内容，全是空的。这个也是空的
![image](https://user-images.githubusercontent.com/17307843/212542035-8da8fc55-9426-45f1-900e-2049b3cce289.png)

**Comment by FrozenFish259** [comment]
没遇到过，用最新版7.4试试

**Comment by 204065248** [comment]
新版的可以，谢了

**Comment by 204065248** [comment]
我知道怎么找基址了，但是我疑惑的是为什么正确的数据还是得不到玩家指针呢
![image](https://user-images.githubusercontent.com/17307843/212543424-a10fa652-ea38-453f-adb6-e9ce2de6d605.png)

**Comment by FrozenFish259** [comment]
正常，说明搜索指针链的途中读到了0值或者就是读取的内存地址非法

**Comment by 204065248** [comment]
ptr_staticFields 这个数据是从哪里得到的呢

**Reply by FrozenFish259** [reply]
逆向dumped数据里面有结构

**Reply by FrozenFish259** [reply]
导入IDA查看结构

**Reply by 204065248** [reply]
![image](https://user-images.githubusercontent.com/17307843/212593653-a8c123e5-4cf4-4413-8f5d-564c82ff1963.png)

我在il2cpp dump出来的内容还是找不到这个数据，不清楚你说的导入IDA是具体是什么意思，我在IDA中查看GameAssembly.dll模块也没有头绪- -

**Reply by FrozenFish259** [reply]
首先这个playersListWithAgoraIDs的字典是在静态区，你得去PlayerController的Class中的静态区查找

**Reply by FrozenFish259** [reply]
![image](https://user-images.githubusercontent.com/49237108/212595157-290b400b-404b-4b7e-8126-df225bb5c298.png)

**Reply by FrozenFish259** [reply]
![image](https://user-images.githubusercontent.com/49237108/212595237-e82ef628-d563-4187-ad2f-03a99995c7d3.png)

**Reply by FrozenFish259** [reply]
PlayerController实例对象 + 0x0 = *PlayerController_Class
PlayerController_Class + 0xB8 = *PlayerController_StaticFileds
PlayerController_StaticFileds + 0x20 = *playersListWithAgoraIDs

**Reply by 204065248** [reply]
这个是怎么搞出来的呢，我的struct里面没这些东西
![image](https://user-images.githubusercontent.com/17307843/212596084-66c65243-bd4f-4315-b4fa-610bac9bea54.png)

**Reply by FrozenFish259** [reply]
Local types

**Reply by 204065248** [reply]
第一次用这个窗口- -。但是我还是没这些结构体，是我少了什么操作吗，我加载的是GameAssembly.dll这个dll

**Reply by FrozenFish259** [reply]
使用脚本导入所有结构

**Reply by FrozenFish259** [reply]
https://github.com/T5ive/Il2CppDumper-GUI

**Reply by 204065248** [reply]
感谢，找到了，但是他执行py脚本的过程中还有报错的内容，这个无关紧要吗
![image](https://user-images.githubusercontent.com/17307843/212601842-ab74a992-9201-438d-b360-66f28add1e39.png)

**Reply by FrozenFish259** [reply]
正常

**Comment by 204065248** [comment]
老哥，你知道这游戏的矩阵是怎么找的不

**Reply by FrozenFish259** [reply]
暂时没有开发绘制

**Reply by FrozenFish259** [reply]
你可以去找关于Camera的类

**Reply by 204065248** [reply]
好的，我研究研究
